File: src/include/asset/AssetSorting.php
<?php

namespace asset;

enum AssetSorting: string
{
	case POPULAR = "popular";
	case LATEST = "latest";
	case OLDEST = "oldest";
	case RANDOM = "random";
	case MOST_CLICKED = "most-clicked";
	case LEAST_CLICKED = "least-clicked";
	case MOST_TAGGED = "most-tagged";
	case LEAST_TAGGED = "least-tagged";
	case OLDEST_VALIDATION_SUCCESS = "oldest-validation-success";
	case LATEST_VALIDATION_SUCCESS = "latest-validation-success";

	/**
	 * Returns the enum value for the string. 
	 * Every other/invalid string gets turned into Sorting::LATEST.
	 */
	public static function fromAnyString(string $string): AssetSorting
	{
		if (in_array($string, array_column(AssetSorting::cases(), 'value'))) {
			return AssetSorting::from($string);
		} else {
			return AssetSorting::LATEST;
		}
	}
}

File: app/asset/asset_sorting.py
from enum import Enum
from typing import Optional


class AssetSorting(str, Enum):
    POPULAR = "popular"
    LATEST = "latest"
    OLDEST = "oldest"
    RANDOM = "random"
    MOST_CLICKED = "most-clicked"
    LEAST_CLICKED = "least-clicked"
    MOST_TAGGED = "most-tagged"
    LEAST_TAGGED = "least-tagged"
    OLDEST_VALIDATION_SUCCESS = "oldest-validation-success"
    LATEST_VALIDATION_SUCCESS = "latest-validation-success"

    def slug(self) -> str:
        return self.value

    @classmethod
    def from_slug(cls, slug: str) -> "AssetSorting":
        return cls.try_from_slug(slug) or cls.LATEST

    @classmethod
    def try_from_slug(cls, slug: str) -> Optional["AssetSorting"]:
        try:
            return cls(slug)
        except ValueError:
            return None
